query/src/main/java/org/apache/kylin/query/util/QueryUtil.java

```diff
@@ -178,6 +178,12 @@ public static String makeErrorMsgUserFriendly(Throwable e) {
                 msg = "NumberFormatException: " + cause.getMessage();
                 break;
             }
+
+            //where in(...) condition has too many elements
+            if (cause.getClass().equals(StackOverflowError.class)) {
```

Comment:
I guess, that **Too Many Filters** is not the only cause of `StackOverflowError`, do you confirm it?

Reply:
You are right. Modify use message